homeassistant/components/light/blinksticklight.py

```diff
@@ -0,0 +1,64 @@
+"""
+homeassistant.components.light.blinkstick
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Support for Blinkstick lights.
+"""
+
+from blinkstick import blinkstick
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+from homeassistant.components.light import (Light, ATTR_RGB_COLOR)
+
+REQUIREMENTS = [ "blinkstick==1.1.7" ]
+DEPENDENCIES = []
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+	stick = blinkstick.find_by_serial( config['serial'])
+
+	add_devices_callback( [BlinkStickLight(stick, config['name'])] )
+
+class BlinkStickLight(Light):
+	def __init__( self, stick, name ):
+		self._stick = stick
+		self._name = name
+		self._serial = stick.get_serial()
+		r,g,b = stick.get_color()
+		self._rgb_color = [r, g, b]
+
+	@property
+	def should_poll(self):
+		return False
+
+	@property
+	def name( self ):
+		return self._name
+
+	@property
+	def rgb_color(self):
+		return self._rgb_color
+
+	@property
+	def is_on(self):
+		r,g,b = self._stick.get_color()
+
+		if r>0 or b>0 or g>0:
+			return True
+		else:
+			return False
+
+	def turn_on(self, **kwargs):
+		""" Turn the device on. """
+
+		if ATTR_RGB_COLOR in kwargs:
+			self._rgb_color = kwargs[ATTR_RGB_COLOR]
```

Comment:
Don't store state, state should come from querying the device using the update method, automatically called by HA.